Feat/aut 1981/hide features

shaveko opened the pull request:
related to https://oat-sa.atlassian.net/browse/AUT-1981

### Description

Authoring features have to be toggable through `services/features` part of config registry

### How to test

Kitchen-playground: 
`https://test-hidefeatures.playground.kitchen.it.taocloud.org/`

Or locally :

Setup the following keys in `config/tao/client_lib_config_registry.conf.php`

```
        'services/features' => array(
            'visibility' => array(

                'taoQtiTest/creator/test/property/timeLimits' => 'hide',
                'taoQtiTest/creator/testPart/property/timeLimits' => 'hide',
                'taoQtiTest/creator/section/property/timeLimits' => 'hide',
                
                'taoQtiTest/creator/testPart/property/itemSessionControl/showFeedback' => 'hide',
                'taoQtiTest/creator/testPart/property/itemSessionControl/allowComment' => 'hide',
                'taoQtiTest/creator/testPart/property/itemSessionControl/allowSkipping' => 'hide',
                
                'taoQtiTest/creator/section/property/itemSessionControl/showFeedback' => 'hide',
                'taoQtiTest/creator/section/property/itemSessionControl/allowComment' => 'hide',
                'taoQtiTest/creator/section/property/itemSessionControl/allowSkipping' => 'hide',

                'taoQtiTest/creator/itemRef/property/itemSessionControl/showFeedback' => 'hide',
                'taoQtiTest/creator/itemRef/property/itemSessionControl/allowComment' => 'hide',
                'taoQtiTest/creator/itemRef/property/itemSessionControl/allowSkipping' => 'hide',
                
                'taoQtiTest/creator/category/presetGroup/warning' => 'hide',

                'taoQtiTest/creator/category/preset/reviewScreen' => 'hide',
                'taoQtiTest/creator/category/preset/markReview' => 'hide',
                'taoQtiTest/creator/category/preset/answerMasking' => 'hide',
                'taoQtiTest/creator/category/preset/areaMasking' => 'hide',
                'taoQtiTest/creator/category/preset/calculator' => 'hide',
                'taoQtiTest/creator/category/preset/calculatorBodmas' => 'hide',
                'taoQtiTest/creator/category/preset/calculatorScientific' => 'hide',
                'taoQtiTest/creator/category/preset/magnifier' => 'hide',
                'taoQtiTest/creator/category/preset/zoom' => 'hide',
                'taoQtiTest/creator/category/preset/apiptts' => 'hide',
            )
        )
```

Check corresponding props are hidden from authoring interface
Enable them by deleting the key or setting the value to `show`

krampstudio commented on views/js/controller/creator/creator.js:
```diff
@@ -66,18 +68,38 @@ define([
     'use strict';
     const logger = loggerFactory('taoQtiTest/controller/creator');
 
+    const filterVisiblePresets = presetGroups => {
```
niptick: can you please document the function

shaveko replied:
Thanks, done

krampstudio commented on views/js/controller/creator/creator.js:
```diff
@@ -24,6 +24,7 @@ define([
     'lodash',
     'helpers',
     'i18n',
+    'services/features',
```
chore: I think we need to update the dependency to tao-core version > 50.10.0 where the service was added : https://github.com/oat-sa/tao-core/releases/tag/v50.10.0  in the composer.json of the extension
And I think we forgot to do it in taoQtiItem too :disappointed:

shaveko replied:
Updated the dependency here and in taoQtiItem as well, please check this PR https://github.com/oat-sa/extension-tao-itemqti/pull/2119

krampstudio reviewed (commented):
It works well. I have a few comments on the source code. Also the tests on the extension aren't passing but it seems it's not due to your changes. Also please check if it makes sense or if it's possible to test your changes (or part of)

shaveko commented on views/js/controller/creator/creator.js:
```diff
@@ -177,84 +199,85 @@ define([
+                            'The test cannot be saved because it currently contains invalid settings.\n' +
+                                'Please fix the invalid settings and try again.'
+                        )
+                    );
```
Thanks, unprettied this line

shaveko commented:
> It works well. I have a few comments on the source code. Also the tests on the extension aren't passing but it seems it's not due to your changes. Also please check if it makes sense or if it's possible to test your changes (or part of)

I've refactored the visibility filtering to separate helper, and added tests

yukipastelcat reviewed (approved):
Code looks good and feature works well, please check the comment regarding log output. Good job👍

- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

meefox reviewed (approved):
Code review LGTM only please remove unecessary console log
